src/CppParser/Parser.cpp

```diff
@@ -1227,15 +1227,18 @@ Parser::WalkTemplateArgumentList(const clang::TemplateArgumentList* TAL,
 {
     using namespace clang;
 
+    auto LocValid = TSTL && !TSTL->isNull();
+    auto LocTypePtrValid = LocValid && TSTL->getTypePtr() != nullptr;
```

Comment:
Please remove != nullptr for simplicity. This way the line will become shorter which would enable you to merge it with the previous one and therefore use just one variable.